bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -655,6 +655,9 @@ public static boolean checkIsType(Type sourceType, Type targetType, List<TypePai
                 if (sourceTypeTag == TypeTags.FINITE_TYPE_TAG) {
                     return isFiniteTypeMatch((BFiniteType) sourceType, targetType);
                 }
+                if (sourceTypeTag == TypeTags.XML_TAG && targetTypeTag == TypeTags.STRING_TAG) {
+                    return ((BXmlType) sourceType).constraint.getTag() == TypeTags.NEVER_TAG;
```

Comment:
Given that this logic is relevant only for `string`, I think we should have the case for `STRING_TAG` separately and move this logic and the `sourceTypeTag == targetTypeTag` check there.